tests/ui_tests/app/page/common/base_page.py

```diff
@@ -0,0 +1,209 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import allure
+from selenium.common.exceptions import (
+    NoSuchElementException,
+    StaleElementReferenceException,
+    TimeoutException,
+)
+from selenium.webdriver.common.keys import Keys
+from selenium.webdriver.remote.webdriver import WebElement
+from selenium.webdriver.support import expected_conditions as EC
+from selenium.webdriver.support.ui import WebDriverWait as WDW
+
+from tests.ui_tests.app.helpers.locator import Locator
+from tests.ui_tests.app.page.common.footer import CommonFooterLocators
+from tests.ui_tests.app.page.common.header import (
+    CommonHeaderLocators,
+    AuthorizedHeaderLocators,
+)
+
+
+class BasePageObject:
+    """BasePageObject is parent class for all ADCM's pages."""
+
+    __slots__ = {"driver", "base_url", "path", "default_wait_timeout"}
+
+    def __init__(self, driver, base_url, path="", default_wait_timeout=5):
+        self.driver = driver
+        self.base_url = base_url
+        self.path = path
+        self.default_wait_timeout = default_wait_timeout
+
+    def open(self, timeout=None):
+        """Open page by its url and path."""
+
+        url = self.base_url + self.path
+        if self.driver.current_url != url:
+            with allure.step(f"Open {url}"):
+                self.driver.get(url)
+        self.wait_url_contains_path(self.path, timer=timeout or self.default_wait_timeout)
+        return self
+
+    def wait_url_contains_path(self, path: str, timer=5) -> None:
+        """Wait url to contain path."""
+
+        WDW(self.driver, timer).until(EC.url_contains(path), message=f"Page with path '{path}' has not been loaded for "
+                                                                     f"{timer} seconds")
+
+    def find_element(self, locator: Locator, time: int = 15) -> WebElement:
+        """Find element on current page."""
+
+        with allure.step(f'Find element "{locator.name}" on page'):
+            return WDW(self.driver, time).until(EC.presence_of_element_located([locator.by, locator.path]),
+                                                message=f"Can't find {locator.name} on page "
+                                                        f"{self.driver.current_url} for {time} seconds")
+
+    def find_elements(self, locator: Locator, time: int = 15) -> [WebElement]:
+        """Find elements on current page."""
+
+        with allure.step(f'Find elements "{locator.name}" on page'):
+            return WDW(self.driver, time).until(EC.presence_of_all_elements_located([locator.by, locator.path]),
+                                                message=f"Can't find {locator.name} on page "
+                                                        f"{self.driver.current_url} for {time} seconds")
+
+    def is_locator_displayed(self, locator: Locator, time: int = 10) -> bool:
+        """Checks if locator is displayed."""
```

Comment:
Hereinafter. We search element, not locator. I think is better if the names of functions will contain `element`, not `locator`. 
It might be worth considering naming `Locator` class as well. It seems that the meaning and content say us that is `Element`

Reply:
renamed methods